src/layout/ProtectedComponents/ProtectedComponents.jsx

```diff
@@ -0,0 +1,21 @@
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { getUserDetails } from '../../redux/reducer/userSlice';
+import { userRoles } from '../../constants/userRoles';
+
+function ProtectedComponents({ children }) {
```

Comment:
@saumier This is like a general function for components.Whenever a components requires role based access, we may pass the corresponding components through this function.There may be changes in the code in the future, depending on the access.
What do you about this?Is this a good approach ?